packages/common-all/src/store/IConfigStore.ts

```diff
@@ -0,0 +1,59 @@
+import { ResultAsync } from "neverthrow";
+import { IDendronError } from "../error";
+import { DendronConfig, DendronConfigValue } from "../types";
+import { DeepPartial } from "../utils";
+
+export type ConfigValue = string | number | object;
+
+export type ConfigReadMode = "override" | "default";
+export type ConfigReadOpts = {
+  mode: ConfigReadMode;
+  useCache?: boolean;
+};
+
+export type ConfigGetOpts = ConfigReadOpts;
+
+export interface IConfigStore {
+  // entire config
+
+  /**
+   * Create a persistant dendron config
+   * If a persistant dendron config exists, return an error
+   */
+  create(
+    defaults?: DeepPartial<DendronConfig>
+  ): ResultAsync<DendronConfig, IDendronError>;
+  /**
+   * Read the entire dendron config
+   */
+  readRaw(): ResultAsync<DeepPartial<DendronConfig>, IDendronError>;
+  read(opts: ConfigReadOpts): ResultAsync<DendronConfig, IDendronError>;
+  /**
+   * Given a dendron config, update the persistant dendron config with the given payload
+   */
+  write(payload: DendronConfig): ResultAsync<DendronConfig, IDendronError>;
+
+  // individual keys
+  /**
+   * Given a dot delimited path, retrieve the config entry value in the persistant dendron config
+   */
+  get(
+    key: string,
+    opts: ConfigGetOpts
+  ): ResultAsync<DendronConfigValue, IDendronError>;
+  /**
+   * Given a dot delimited path, update the config entry value with given value in the persistant dendron config
+   * returns previous value
+   */
+  update(
+    key: string,
+    value: DendronConfigValue
+  ): ResultAsync<DendronConfigValue | undefined, IDendronError>;
+  /**
+   * Given a dot delimited path, delete the config entry in the persistant dendron config
```

Comment:
nit on comment: why 'dot delimited path' - shouldn't it just be any path?

Reply:
Sorry the docstring is a bit confusing

I'm talking about the path arg you pass into `_.get()`: https://lodash.com/docs/4.17.15#get

I'll add some examples to make that clearer